crates/typst/src/geom/scalar.rs

```diff
@@ -4,7 +4,38 @@ use super::*;
 ///
 /// Panics if it's `NaN` during any of those operations.
 #[derive(Default, Copy, Clone)]
-pub struct Scalar(pub f64);
+pub struct Scalar(f64);
+
+// We have to detect NaNs this way since `f64::is_nan` isn’t const
+// on stable yet:
+// ([tracking issue](https://github.com/rust-lang/rust/issues/57241))
+#[allow(clippy::unusual_byte_groupings)]
+const fn is_nan_const(x: f64) -> bool {
+    let x_bits = unsafe { std::mem::transmute::<f64, u64>(x) };
```

Comment:
A safety comment that also explains that `to_bits` is not const fn would be nice.